### PR TITLE
Attempt to install dependencies with npm --legacy-peer-deps for nextjs

### PR DESCRIPTION
**Pull request by andreia-oca**
# Pull Request Template

## Type of change

-   [ ] 🍕 New feature
-   [ ] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [x] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Description

This PR:
* automatically install dependencies for nextjs projects
* adds a retry step for installing dependencies with `--legacy-peer-deps`
* throws friendly errors for common npm errors such as `invalid package.json` , `package not found`, and `missing package version`.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [ ] I have added tests;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/commands/deploy/nextjs/deploy.ts [hunk 1]
```diff
@@ -39,22 +42,43 @@ import { computeAssetsPaths } from "./assets.js";
...
+    scripts?: {
+        deploy: string;
+        build?: string;
+        start?: string;
+    };
+    environment?: {
+        [key: string]: string;
+    };
+    subdomain?: string;
+};
```
Comment (costinsin) [inline src/commands/deploy/nextjs/deploy.ts:60]: Looks like this is more of a SSRFramework, than specific to NextJS

**Review by costinsin — APPROVED** [review]
LGTM

File: src/commands/deploy/nextjs/deploy.ts (hunk 1 above)
Comment (andreia-oca) [reply src/commands/deploy/nextjs/deploy.ts]: You are right, I have updated it and move it on a higher level.

File: src/commands/deploy/utils.ts [hunk 2]
```diff
@@ -124,6 +125,67 @@ export async function getOrCreateEmptyProject(
     return { projectId: project.id, projectEnvId: projectEnv.id };
 }
 
+export async function attemptToInstallDependencies(args: string[] = [], currentPath: string) {
```
Comment (vladiulianbogdan) [inline src/commands/deploy/utils.ts:128]: no return type

**Review by vladiulianbogdan — APPROVED** [review]

**Pull request merged**